server/api/cart.js

```diff
@@ -0,0 +1,84 @@
+const router = require("express").Router();
+const {
+  models: { Order, User, LineItem, Product },
+} = require("../db");
+module.exports = router;
+
+const isLoggedIn = async (req, res, next) => {
+  try {
+    req.user = await User.findByToken(req.headers.authorization);
+    next();
+  } catch (ex) {
+    next(ex);
+  }
+};
```

Comment:
likewise, you can create an `isAdmin` middleware to check whether the user is an admin